default/scripting/game_rules.focs.py

```diff
@@ -6,6 +6,7 @@
     default=1.0,
     min=0.1,
     max=10.0,
+    rank=310,
```

Comment:
This is Python, let's not hardcode any numbers. 

If the import is not available, just copy the code https://docs.python.org/3/library/itertools.html#itertools.count
```python
from itertools import count

"""
The order of rules defines their appearance in the UI.
"""
counter = count(start = 310)
...
rank=next(counter),
...
rank=next(counter),

```

Reply:
i thought about ranking them by order of appearance in the file but problem is there are also the rules you specify on the c++ side. And that is not that obvious at all what's their order, plus both sets ultimately go into one collection. An explicit rank allows to deal with both types of rules, and also allows ordering of categories without splitting that into yet another layer

Reply:
Got it.  

In that case, you could use different counters.  Python Rules that use the same counter will be processed in order of declaration.

Reply:
would be fine if categories between itnernal C++ game rules and the ones defined on python side did not share categories, and you could have a counter for each category then. But there are shared categories. 

I suppose the counter could go up like by 10 or 100 or something so there's space to insert internal game rules into the same category where you want, but then you still need to actually know the rank of python-declared game rules to know what to put on the c++ side so it ends up inbetween, kinda defeating the purpose of auto-generating ranks on python side, I would think

Reply:
ok so I've added a counter like you suggested. itertools import was not available so I defined it in the file. Set step to 100 so there's room to squeeze internal engine rules.